cas-server-support-jdbc/src/main/java/org/jasig/cas/services/JpaServiceRegistryDaoImpl.java

```diff
@@ -35,6 +37,7 @@
     @PersistenceContext
     private EntityManager entityManager;
 
+    @Transactional(readOnly = false)
```

Comment:
I had a talk with one of my colleagues who understands databases and hibernate really well, and we both came to the conclusion that having that a Transactional at the top of the method doesn't really get us anything. There will be 2 instead of 1, one to achieve true synchronization and locking, it will require some truly fancy hibernate heroics and it will entirely depend on the database of choice.